packages/common-ts/src/base-service/validators.ts

```diff
@@ -39,6 +41,14 @@ const wallet = makeValidator<Signer>((input) => {
   }
 })
 
+const logLevel = makeValidator<LogLevel>((input) => {
+  if (!logLevels.includes(input as LogLevel)) {
+    throw new Error(`expected log level to be one of ${logLevels}`)
+  } else {
+    return input as LogLevel
+  }
+})
```

Comment:
This is way cleaner with zod.  
Zod is generally useful.  Anytime you are taking unvalidated input and trying to validate it, zod is the best way to do it.   
``` typescript
const logLevelValidator =  z.union([
  z.literal('trace'), 
  z.literal('debug'),
  z.literal('info'),
  z.literal('warn'),
  z.literal('error'),
  z.literal('fatal'),
]);
// logLevel is typesafe now and will throw an error if it's not right type
const logLevel = logValidator.parse(input)

```

Reply:
Interesting, gotta check that out for later